source/slang/slang-check-decl.cpp

```diff
@@ -7981,19 +8326,325 @@ namespace Slang
             }
             if (seqStmtChild->stmts.getCount() == 0)
                 continue;
-            seqStmt->stmts.insert(insertOffset, seqStmtChild);
+
+            auto seqStmt = _ensureCtorBodyIsSeqStmt(m_astBuilder, ctor);
+            seqStmt->stmts.insert(ctorInfo.m_insertOffset++, seqStmtChild);
+        }
+
+        // Note: we assume only 1 inheritance decl currently.
+        // Pre-calculate if we have a base-type and its associated ctor-list
+        // We also must ensure inheritance-decl is checked, else we may not have up-to-date 'DerivativeMemberAttribute' modifiers
+        DeclAndCtorInfo* baseStructInfo = nullptr;
+        for(auto& i : inheritanceInfoList)
+        {
+            ensureDecl(i.m_inheritanceDecl, DeclCheckState::VarInitExprAreChecked);
+            if(as<StructDecl>(i.m_inheritanceBaseDecl))
+                baseStructInfo = &i;
+        }
+
+        // pre-calculate any requirements of a CudaHostAttribute
+        HashSet<VarDeclBase*> requiresCudaHostModifier;
```

Comment:
It doesn't seem to worth this much complexity for such a corner case. We need to get rid of all the TorchTensorType` logic here, and just insert the cuda host decoration in a separate legalization pass when compiling to pytorch.